pkg/ccl/backupccl/restore_span_covering.go

```diff
@@ -154,6 +154,24 @@ func makeSimpleImportSpans(
 	for _, requiredSpan := range requiredSpans {
 		filteredSpans := filter.filterCompleted(requiredSpan)
 		for _, span := range filteredSpans {
+			// endKeyNotCoveredFiles is a collection of files that, due to the end key
+			// inclusive nature of their spans, do not have their end key covered by
+			// the current cover. These are kept around so that they can be included
+			// in the next entry's file list whenever the cover is extended. This
+			// collection is populated from files in two cases:
+			//
+			//  1. A file has an end key equal to the end key of the last cover
```

Comment:
i've read through this case like 5 times and I still don't fully understand it. Is it possible to add a quick example or 2 in the comment?

Reply:
Another way to restructure this comment: I see 4 call sites of drain. It might worth commenting an example for each.

Reply:
I added an example to this comment